plugins/destination/gremlin/client/spec.go

```diff
@@ -91,3 +109,108 @@ func (s *Spec) Validate() error {
 
 	return nil
 }
+
+func (Spec) JSONSchemaExtend(sc *jsonschema.Schema) {
+	authMode := sc.Properties.Value("auth_mode")
+	authMode.Enum = []any{authModeNone, authModeBasic, authModeAWS}
```

Comment:
IMO this should be just done in the `authMode` type schema. Options:
* Implement `JSONSchema` method for `authMode` that will just return the schema for the type
* Implement `JSONSchemaExtend` method for `authMode`

See example in CLI: https://github.com/cloudquery/cloudquery/blob/main/cli/internal/specs/v0/pk_mode.go#L47